apps/multiplatform/common/src/commonMain/kotlin/chat/simplex/common/views/chat/ChatView.kt

```diff
@@ -961,21 +989,104 @@ fun BoxScope.ChatItemsList(
     }
   }
 
-  PreloadItems(chatInfo.id, listState, ChatPagination.UNTIL_PRELOAD_COUNT, loadPrevMessages)
-
   Spacer(Modifier.size(8.dp))
   val reversedChatItems = remember { derivedStateOf { chatModel.chatItems.asReversed() } }
+  val revealedItems = rememberSaveable { mutableStateOf(setOf<Long>()) }
+  val sections by remember { derivedStateOf { reversedChatItems.value.putIntoSections(revealedItems.value) } }
```

Comment:
try to exclude from you practice using ```val something by ...```. Because when you do this, you cannot safely use that value inside of other Unit blocks because in that Unit blocks that value you're pointing at can be stale. In other words, you're pointing to a value that was already exist in time of creation of such block of code and will not change until next recomposition. And if you send such block of code to external function, the value will be old there. Instead, use this way:

```val something = remember {```

This means that in other Unit blocks you will get a value by calling ```something.value``` and it will be actual.

Also, another advantage of doing so is the fact that compose will not need to recompose containing function on change of value inside remember. So, for example, when you have:

```kotlin
val something = remember { derivedStateOf { chatModel.chats.value } }
```

and then you use ```something,value``` only inside NON-@Composable function, Compose will give you actual value but will not recompose function that contains this declaration which prevents unneeded recompositions.